data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -317,25 +343,25 @@ then
 	touch cco-bootstrap.done
 fi
 
-# Wait for the etcd cluster to come up.
-until bootkube_podman_run \
-		--rm \
-		--name etcdctl \
-		--env ETCDCTL_API=3 \
-		--volume /opt/openshift/tls:/opt/openshift/tls:ro,z \
-		--entrypoint etcdctl \
-		"${MACHINE_CONFIG_ETCD_IMAGE}" \
-		--dial-timeout=10m \
-		--cacert=/opt/openshift/tls/etcd-ca-bundle.crt \
-		--cert=/opt/openshift/tls/etcd-client.crt \
-		--key=/opt/openshift/tls/etcd-client.key \
-		--endpoints="${ETCD_ENDPOINTS}" \
-		endpoint health
-do
-	echo "etcdctl failed. Retrying in 5 seconds..."
-	sleep 5
-done
+# in case of single node, if we removed etcd, there is no point to wait for it on restart
+if [ "$BOOTSTRAP_INPLACE" = true ]
+then
+    if [ ! -f stop-etcd.done ]
+    then
+        # Wait for the etcd cluster to come up.
+        wait_for_etcd_cluster
+    fi
+else
+    wait_for_etcd_cluster
+fi
```

Comment:
nit: this could be flattened... if bootstrap_inpace = false or ! -f stop-etcd.done; then wait_for_etcd_cluster; fi